staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -193,6 +211,64 @@ func (c *threadSafeMap) ByIndex(indexName, indexedValue string) ([]interface{},
 	return list, nil
 }
 
+// ByIndexes Joint indexing by multiple indexers while supporting Operator
+func (c *threadSafeMap) ByIndexes(conds IndexConditions) ([]interface{}, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	set := sets.NewString()
+	for _, cond := range conds {
+		populatedIndex, err := c.getIndex(cond)
+		if err != nil {
+			return nil, err
+		}
+		if set.Len() == 0 {
+			set = populatedIndex
+			continue
+		}
+		set = set.Intersection(populatedIndex)
+	}
+
+	list := make([]interface{}, 0, set.Len())
+	for key := range set {
+		list = append(list, c.items[key])
+	}
+
+	return list, nil
+}
+
+// getIndex can get Index by given IndexCondition
+func (c *threadSafeMap) getIndex(cond IndexCondition) (sets.String, error) {
+	indexFunc := c.indexers[cond.IndexName]
+	if indexFunc == nil {
+		return nil, fmt.Errorf("Index with name %s does not exist", cond.IndexName)
+	}
+
+	index := c.indices[cond.IndexName]
+
+	set := index[cond.IndexedValue]
+
+	populatedIndex, err := c.populateIndex(cond.Operator, set)
+	if err != nil {
+		return nil, err
+	}
+
+	return populatedIndex, nil
+}
+
+// populateIndex would populate real Index according to Operator
+func (c *threadSafeMap) populateIndex(op selection.Operator, set sets.String) (sets.String, error) {
```

Comment:
nit: same here. I would avoid overload the meaning of "Index". How about `applyOperator`?